### PR TITLE
Dimpt 1 increase readability maintainability

### PR DESCRIPTION
**Pull request by aclseelen**

### CODE REVIEW

File: src/main/java/org/folio/rest/impl/Processor.java [hunk 1]
```diff
@@ -429,16 +442,16 @@ private String processRules(String data, JsonArray rules, Leader leader){
     //there are rules associated with this subfield / control field - to instance field mapping
     String originalData = data;
     for (int ruleIndex = 0; ruleIndex < rules.size(); ruleIndex++) {
-      ProcessedRule pr = processRule(rules.getJsonObject(ruleIndex), leader, data, originalData);
-      data = pr.getData();
-      if (pr.doBreak()) {
+      ProcessedSingleItem psi = processRule(rules.getJsonObject(ruleIndex), leader, data, originalData);
+      data = psi.getData();
+      if (psi.doBreak()) {
```
Comment (julianladisch) [inline src/main/java/org/folio/rest/impl/Processor.java]: Wrapping data and doBreak in ProcessedSingleItem is more complicated than directly returning data and using `java.util.concurrent.atomic.AtomicBoolean doBreak`:
```
data = processRule(rules.getJsonObject(ruleIndex), leader, data, originalData, doBreak);
if (doBreak.get()) {
  break;
}
```

File: src/main/java/org/folio/rest/service/ProcessorHelper.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+import io.vertx.core.json.JsonObject;
+
+public class ProcessorHelper {
+
+  private ProcessorHelper() {}
+
+  public static String[] getFunctionsFromCondition(JsonObject condition) {
+    return condition.getString("type").split(",");
+  }
```
Comment (julianladisch) [inline src/main/java/org/folio/rest/service/ProcessorHelper.java]: If you trim() each String here you don't need to do it multiple times afterwards.

File: src/main/java/org/folio/rest/impl/Processor.java [hunk 3]
```diff
@@ -479,68 +492,97 @@ private ProcessedRule processRule(JsonObject rule, Leader leader, String data, S
...
-      //may not be the first function listed in the function list
-      //a little wasteful, but this will probably only loop at most over 2 or 3 function names
-      for (String function : functions) {
-        if(CUSTOM.equals(function.trim())){
-          isCustom = true;
-          break;
-        }
-      }
+      String[] functions = ProcessorHelper.getFunctionsFromCondition(condition);
+      isCustom = checkIfAnyFunctionIsCustom(functions, isCustom);
```
Comment (julianladisch) [inline src/main/java/org/folio/rest/impl/Processor.java]: Writing it this way shows more clearly what happens:
`isCustom = isCustom || hasCustomFunction(functions)`

File: src/main/java/org/folio/rest/impl/Processor.java [hunk 4]
```diff
@@ -479,68 +492,97 @@ private ProcessedRule processRule(JsonObject rule, Leader leader, String data, S
...
-      }
-      if(!conditionsMet){
-        //all conditions for this rule we not met, revert data to the originalData passed in.
-        data = originalData;
-        break;
-      }
+      ProcessedSinglePlusConditionCheck processedCondition =
+        processCondition(condition, data, originalData, leader, conditionsMet, ruleConstVal, isCustom);
+      data = processedCondition.getData();
+      conditionsMet = processedCondition.isConditionsMet();
```
[the remark this answers is not in the document]
Comment (aclseelen) [reply src/main/java/org/folio/rest/impl/Processor.java]: I altered this whole struct design into the boolean reference design for all cases. However, I just reverted all the changes by the insight that we're talking about methods that are eventually each other's sub-method. I considered the doBreak AtomicBoolean too dangerous to be used in multiple nested loops. Best option was to revert everything and keep the pojos. I might still want to do this in a later stage, but I am not sure enough now.

File: src/main/java/org/folio/rest/service/ProcessorHelper.java (hunk 2 above)
Comment (aclseelen) [reply src/main/java/org/folio/rest/service/ProcessorHelper.java]: I just moved existing code to another place. I see your point, but the easy solution is not available in the String class. I would have to iterate over the String[] to trim() afterwards. It's not the moment to decide whether such is what I actually want. I don't want to alter functionality in this stage, unless all the consequences are absolutely clear.

**Pull request merged**